### PR TITLE
Only deregister once

### PR DESCRIPTION
**Pull request by jonmv**
@hakonhall or @freva please review and merge if this is an OK assumption, i.e., stop only needs to be called once per agent in this test.

### CODE REVIEW

**Comment by hakonhall** [comment]
stopForHostSuspension is idempotent and may be called multiple times.  It will remove the container of the associated node, and is a no-op if it is already removed.

File: node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/integration/ContainerTester.java [hunk 1]
```diff
@@ -92,13 +93,16 @@ public class ContainerTester implements AutoCloseable {
...
+                    AtomicBoolean stopped = new AtomicBoolean();
                     @Override public void converge(NodeAgentContext context) {
                         super.converge(context);
                         phaser.arriveAndAwaitAdvance();
                     }
                     @Override public void stopForHostSuspension(NodeAgentContext context) {
-                        super.stopForHostSuspension(context);
-                        phaser.arriveAndDeregister();
+                        if (stopped.compareAndSet(false, true)) {
+                            super.stopForHostSuspension(context);
```
Comment (hakonhall) [inline node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/integration/ContainerTester.java:103]: So `super.stopForHostSuspension()` should be called for each invocation, but phaser.arriveAndDeregister() only on the first invocation per node?

Comment (jonmv) [reply node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/integration/ContainerTester.java:103]: Ideally they would just call `arriveAndAwaitAdvance` on each tick, but I believe these agents may be removed, and no longer tick, is that right? In that case, we need to deregister ...

Comment (jonmv) [reply node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/integration/ContainerTester.java:103]: Hmm, perhaps the problem is rather that they `tick` after being removed here ... Urgh.

Comment (freva) [reply node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/integration/ContainerTester.java:103]: Not sure why we have deregister on `stopForHostSuspension()`. If you want it at as the last call, put it inside `stopForRemoval()`. After that `tick()` will never be called again.

Comment (jonmv) [reply node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/integration/ContainerTester.java]: Yes, I must have thought suspension was removal when I added this synchronisation some time back. It looks right to just arrive as usual when stopping for suspension. Done now.

Comment (hakonhall) [reply node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/integration/ContainerTester.java]: Node agent instances are normally preserved with stopForHostSuspension() and until jdisc deconstruction.  Only under very particular circumstances would a node agent may be removed after it has invoked stopForHostSuspension(), something like: the reason for the reboot no longer exists, and a has been removed from an application. E.g. RebootTest doesn't test for this.

**Review by hakonhall — APPROVED** [review]

**Pull request merged**